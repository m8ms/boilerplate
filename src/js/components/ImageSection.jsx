import React from 'react';

class ImageSection extends React.Component {


    constructor(props) {
        super(props);
    }

    getCurrentImageLink(){
        return this.props.ImagesStore.currentImage && this.props.ImagesStore.currentImage.link;
    }

    getCurrentImageTitle(){
        return this.props.ImagesStore.currentImage && this.props.ImagesStore.currentImage.title;
    }

    render() {
        console.log(this.props);
        return (
            <div className="image-section">
                <h4>{this.getCurrentImageTitle()}</h4>
                <img className="image-section__image" src={this.getCurrentImageLink()}/>
            </div>

        )
    }
}

export default ImageSection;
