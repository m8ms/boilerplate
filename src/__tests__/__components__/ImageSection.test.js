import React from 'react';
import ImageSection from '../../js/components/ImageSection';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

const item = {
    id: 'GRDXXdF',
    title: null,
    description: 'Sorry I haven\'t been uploading stuff like this lately. Hopefully this\'ll make up for it.',
    datetime: 1523224179,
    type: 'image/png',
    animated: false,
    width: 6000,
    height: 7000,
    size: 263782,
    views: 3965,
    tags: [],
    link: 'https://i.imgur.com/GRDXXdF.png',
    ups: 34,
    downs: 2,
    points: 32,
    score: 2
};

it('renders correctly with no item', () => {
    const tree = renderer
        .create(<ImageSection/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with item', () => {

    const tree = renderer
        .create(<ImageSection item={item}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with modifiers', () => {

    const tree = renderer
        .create(<ImageSection item={item} modifiers="large single"/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
