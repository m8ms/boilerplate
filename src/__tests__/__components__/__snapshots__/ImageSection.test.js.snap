// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with item 1`] = `
<div
  className="image-section"
>
  <div
    className="image-section__image-wrap"
  >
    <img
      className="image-section__image"
      src="https://i.imgur.com/GRDXXdF.png"
    />
  </div>
  <div
    className="image-section__image-meta"
  >
    <div
      className="image-section__title"
    />
    <div
      className="image-section__description"
    >
      Sorry I haven't been uploading stuff like this lately. Hopefully this'll make up for it.
    </div>
    <div
      className="image-section__views-count"
    >
      3965
       views
    </div>
    <div
      className="image-section__date"
    >
      April 8, 2018 11:49 PM
    </div>
  </div>
</div>
`;

exports[`renders correctly with modifiers 1`] = `
<div
  className="image-section"
>
  <div
    className="image-section__image-wrap"
  >
    <img
      className="image-section__image"
      src="https://i.imgur.com/GRDXXdF.png"
    />
  </div>
  <div
    className="image-section__image-meta"
  >
    <div
      className="image-section__title"
    />
    <div
      className="image-section__description"
    >
      Sorry I haven't been uploading stuff like this lately. Hopefully this'll make up for it.
    </div>
    <div
      className="image-section__views-count"
    >
      3965
       views
    </div>
    <div
      className="image-section__date"
    >
      April 8, 2018 11:49 PM
    </div>
  </div>
</div>
`;

exports[`renders correctly with no item 1`] = `
<div
  className="image-section"
>
  <div
    className="image-section__image-wrap"
  >
    <img
      className="image-section__image"
      src={undefined}
    />
  </div>
  <div
    className="image-section__image-meta"
  >
    <div
      className="image-section__title"
    />
    <div
      className="image-section__description"
    />
    <div
      className="image-section__views-count"
    >
       views
    </div>
    <div
      className="image-section__date"
    >
      
    </div>
  </div>
</div>
`;
